build.py

```diff
@@ -94,6 +92,45 @@ def __init__(self, name, params, dut):
         self.top = args.top
         # the tests parameters
         self.params = params # FIXME ABD
+
+
+        # Load memory configuration from JSON file or use defaults
+        self.load_memory_config()
+
+    def load_memory_config(self):
+        # Default JSON file location
+        json_directory = 'app/cfg/'
+
+        # Check if the -f flag is provided
+        if args.f:
+            json_file = os.path.join(json_directory, args.f +'.json')
+        else:
+            # If not provided, use a default JSON file located in /app/cfg
+            json_file = os.path.join(json_directory, 'default.json')
+
+        try:
+            # Try to load memory configuration from the specified JSON file
+            with open(json_file) as config_file:
+                config_data = json.load(config_file)
+
+            # Update memory offsets and lengths if they exist in the JSON file
+            Test.I_MEM_OFFSET = str(config_data.get('I_MEM_OFFSET', 0x00000000))
+            Test.I_MEM_LENGTH = str(config_data.get('I_MEM_LENGTH', 0x00010000))
+            Test.D_MEM_OFFSET = str(config_data.get('D_MEM_OFFSET', 0x00010000))
+            Test.D_MEM_LENGTH = str(config_data.get('D_MEM_LENGTH', 0x0000F000))
+
+        except FileNotFoundError:
+            # If the JSON file does not exist, use default values
+            print_message(f'[INFO] Using default memory configuration for {self.name}')
+            Test.I_MEM_OFFSET = str(0x00000000)
```

Comment:
If there is no "-cfg" ("-f" flag, we should point to a default cfg file - not use hard coded values from the script.

Reply:
will be done!

Reply:
Please "resolve conversation" when there are no more question so we can track what are the "open" comments and which are conversations are "resolved"